### ISSUE TITLE
sql: clarify syntax error with context information

### ISSUE THREAD

**Issue by erichocean** [cockroachdb/cockroach#25332]
bug report

`select max(id) from skus where last_processed_at > '2018-05-04 04:50:42.256966+00:00' order by id asc;`

`pq: column "id" must appear in the GROUP BY clause or be used in an aggregate function`

As you can see, `id` definitely appears in an aggregate function.

(edit) blocked on #2037.

**Comment by erichocean** [comment]
I'm running Cockroach 2.0, and this error appeared in the `cockroach sql` client.

**Comment by rytaft** [comment]
Hi @erichocean, as we discussed on the Gitter, this is working as expected. The error is due to the fact that `id` is contained in the `order by` clause without being a `group by` column. Since the query will return exactly one row, the `order by` clause is not necessary, and the query will work if you just remove the `order by` clause.

But as you pointed out, there is another issue -- the error message is confusing since `id` is in fact used in an aggregate function. This is the same error message that PostgreSQL returns for this error, which is probably why we chose it. However, it may be worth diverging from PostgreSQL in this case in order to reduce confusion. It would be better to just say, `pq: column "id" must appear in the GROUP BY clause`, and remove the second half of the error message.

Leaving this open for discussion.

**Comment by rytaft** [comment]
cc @knz

**Comment by knz** [comment]
I think the main error message is fine, consider for example the query `select max(k), v from kv`, the error here would be to use `v` outside of an aggregate function. 

However what pg has and CockroachDB doesn't, is *syntactic context* for the error message, where the error message would tell you where in the query the error was found. This is in effect the problem we would solve by addressing #2037, one of our longest open issues. It will be addressed.

**Comment by knz** [comment]
> It would be better to just say, pq: column "id" must appear in the GROUP BY clause, and remove the second half of the error message.

Also a valid correction for the query is `ORDER BY max(id)`, so placing `id` in the GROUP BY is not the only solution
